test/common-test.js

```diff
@@ -43,9 +43,20 @@ module.exports = {
         });
     },
 
-    createComponent: function(getComponent) {
+    createComponent: function(factory) {
 
-        var component = getComponent();
+        var node;
+        var metadata = {
+            facade: function(facade) {
+                node = facade;
+            }
+        };
+        var getComponent = _.isFunction(factory) ? factory : factory.getComponent;
```

Comment:
Why is factory arriving sometimes a a function and sometimes as an object?  Maybe a comment would help.